### PR TITLE
To implement new layouts related to 9.1.6ch

### PR DESCRIPTION
**Pull request by sunghee-hwang**
***
<a href="https://pr-preview.s3.amazonaws.com/AOMediaCodec/iamf/pull/834.html" title="Last updated on Jul 1, 2024, 6:53 AM UTC (c136970)">Preview</a> | <a href="https://pr-preview.s3.amazonaws.com/AOMediaCodec/iamf/834/843be0e...c136970.html" title="Last updated on Jul 1, 2024, 6:53 AM UTC (c136970)">Diff</a>

### CODE REVIEW

File: index.bs [hunk 1]
```diff
@@ -1012,35 +1010,55 @@ In this version of the specification, [=expanded_loudspeaker_layout=] indicates
...
+
+For a given input [=3D audio signal=] with an expanded channel layout defined in [=expanded_loudspeaker_layout=], [=num_layers=] SHALL be set to 1 (i.e., it is a non-scalable channel audio element). Except [=9.1.6ch=] [=Audio Element=], it is RECOMMENDED to use such an [=Audio Element=] as an auxiliary [=Audio Element=] to be mixed with a primary [=Audio Element=] (e.g., TOA or 7.1.4ch) within a [=Mix Presentation=]. If parsers encounter a [=loudspeaker_layout=] = 15 for any layer other than the first layer, they SHOULD skip the [=channel_audio_layer_config=] for that layer and all subsequent layers.
+
+Stereo (L/R) is shared with the stereo pair FLc/FRc, Stereo-RS (Lrs/Rrs) is shared with the stereo pair BL/BR, Stereo-TF (Ltf/Rtf) is shared with the stereo pair TpFL/TpFR, Stereo-TB (Ltb/Rtb) is shared with the stereo pair TpBL/TpBR and 3.0ch (L/C/R) is shared with FLc/FC/FRc.
```
Comment (felicialim) [inline index.bs:1060]: Could you please clarify what "shared with" means?

File: index.bs [hunk 2]
```diff
@@ -2418,12 +2436,15 @@ In this section, for a given x.y.z layout, the next highest layout x'.y'.z' mean
 
 This section defines the renderer to use, given a channel-based [=Audio Element=] and a loudspeaker playback layout.
 
+<dfn noexport>22.2ch</dfn> represents [=Loudspeaker configuration for Sound System H (9+10+3)=].
```
Comment (felicialim) [inline index.bs:2439]:
```suggestion
<dfn noexport>22.2ch</dfn> represents the [=Loudspeaker configuration for Sound System H (9+10+3)=].
```

File: index.bs [hunk 3]
```diff
@@ -2751,6 +2776,76 @@ The 3.1.2ch down-mix matrix for 7.1.4ch is given below, where \(p = 0.707\).
 \end{bmatrix}
 \]
 
+The 9.1.6ch down-mix matrix for 22.2ch is given below, where \(p = 0.707\) and \(q = 0.5\). This down-mix matrix is generated based on Section 8.1 and Table 16 of [[!ITU-2127-0]]
```
Comment (felicialim) [inline index.bs:2779]:
```suggestion
The 9.1.6ch down-mix matrix for 22.2ch is given below, where \(p = 0.707\) and \(q = 0.5\). This down-mix matrix is generated based on Section 8.1 and Table 16 of [[!ITU-2127-0]].
```

**Review by felicialim — COMMENTED** [review]
nit: we use both "Center" channel and "Centre" channel in the spec - we should probably just stick to one spelling for consistency.

File: index.bs (hunk 1 above)
Comment (sunghee-hwang) [reply index.bs:1060]: For example, for 9.1.6ch + FLc/FRc, the actual signaling would be 9.1.6ch + L/R.

**Comment by sunghee-hwang** [comment]
> nit: we use both "Center" channel and "Centre" channel in the spec - we should probably just stick to one spelling for consistency.

Let's handle it with the other editorial updates.

File: index.bs (hunk 1 above)
Comment (felicialim) [reply index.bs:1060]: Got it. WDYT about making it more explicit with something like "The following channel layouts may additionally be indicated using an existing loudspeaker_layout or expanded_loudspeaker_layout. The stereo pair FLc/FRc is indicated using Stereo (L/R)."

Comment (sunghee-hwang) [inline index.bs:1060]:
```suggestion
The following channel layouts may additionally be indicated using an existing loudspeaker_layout or expanded_loudspeaker_layout.  The stereo pair FLc/FRc is indicated using Stereo (L/R), the stereo pair BL/BR is indicated using Stereo-RS (Lrs/Rrs), the stereo pair TpFL/TpFR is indicated using Stereo-TF (Ltf/Rtf), the stereo pair TpBL/TpBR is indicated using Stereo-TB (Ltb/Rtb), and FLc/FC/FRc is indicated using 3.0ch (L/C/R).
```

Comment (sunghee-hwang) [reply index.bs]: Thank you for the improvement.

File: index.bs [hunk 4]
```diff
@@ -1012,35 +1010,55 @@ In this version of the specification, [=expanded_loudspeaker_layout=] indicates
...
-  <td>1</td><td>Stereo-S</td><td>Ls/Rs</td><td>The surround subset (Ls/Rs) of 5.1.4ch</td>
+  <td>4</td><td>Stereo-TF</td><td>Ltf/Rtf</td><td>The top front subset (Ltf/Rtf) of [=7.1.4ch=]</td>
 </tr>
 <tr>
-  <td>2</td><td>Stereo-SS</td><td>Lss/Rss</td><td>The side surround subset (Lss/Rss) of 7.1.4ch</td>
+  <td>5</td><td>Stereo-TB</td><td>Ltb/Rtb</td><td>The top back subset (Ltb/Rtb) of [=7.1.4ch=]</td>
 </tr>
 <tr>
-  <td>3</td><td>Stereo-RS</td><td>Lrs/Rrs</td><td>The rear surround subset (Lrs/Rrs) of 7.1.4ch</td>
+  <td>6</td><td>Top-4.0ch</td><td>Ltf/Rtf/Ltb/Rtb</td><td>The top 4 channels (Ltf/Rtf/Ltb/Rtb) of [=7.1.4ch=]</td>
```
Comment (sunghee-hwang) [inline index.bs:1031]:
```suggestion
  <td>6</td><td>Top-4ch</td><td>Ltf/Rtf/Ltb/Rtb</td><td>The top 4 channels (Ltf/Rtf/Ltb/Rtb) of [=7.1.4ch=]</td>
```

**Comment by sunghee-hwang** [comment]
[Question] How do you feel adding the Top-6ch of 9.1.6ch?
As we agreed on having the Top-4ch of 7.1.4ch, it makes sense to have the Top-6ch of 9.1.6ch.

**Comment by felicialim** [comment]
> [Question] How do you feel adding the Top-6ch of 9.1.6ch? As we agreed on having the Top-4ch of 7.1.4ch, it makes sense to have the Top-6ch of 9.1.6ch.

This makes sense.

File: index.bs [hunk 5]
```diff
@@ -1012,35 +1010,55 @@ In this version of the specification, [=expanded_loudspeaker_layout=] indicates
...
 <tr>
-  <td>7</td><td>3.0ch</td><td>L/C/R</td><td>The front 3 channels (L/C/R) of 7.1.4ch</td>
+  <td>10</td><td>Stereo-Si</td><td>SiL/SiR</td><td>The side subset (SiL/SiR) of [=9.1.6ch=]</td>
 </tr>
 <tr>
-  <td>8 ~ 255</td><td>Reserved</td><td></td><td></td>
+  <td>11</td><td>Stereo-TpSi</td><td>TpSiL/TpSiR</td><td>The top side subset (TpSiL/TpSiR) of [=9.1.6ch=]</td>
+</tr>
+<tr>
+  <td>12 ~ 255</td><td>Reserved</td><td></td><td></td>
```
Comment (sunghee-hwang) [inline index.bs:1049]:
```suggestion
<tr>
  <td>12</td><td>Top-6ch</td><td>TpFL/TpFR/TpSiL/TpSiR/TpBL/TpBR</td><td>The top 6 channels (TpFL/TpFR/TpSiL/TpSiR/TpBL/TpBR) of [=9.1.6ch=]</td>
</tr>
<tr>
  <td>13 ~ 255</td><td>Reserved</td><td></td><td></td>
```

**Comment by jingbo-marquis** [comment]
4.3. IA Base-Enhanced Profile
They SHALL be able to handle up to 28 channels.

    The 28 channels limit applies to the sum of channels across all [Audio Element ]s in a [Mix Presentation ])before mixing.
    One example is a mix with 3rd-order Ambisonics (16 channels) + 7.1.4ch (12 channels).


if 9.1.6 layout is added as input layout , the max channels in a mix should be 32 which 16 channel is for 3rd-order Ambisonics and 16 channels are for 9.1.6

**Comment by sunghee-hwang** [comment]
> 4.3. IA Base-Enhanced Profile They SHALL be able to handle up to 28 channels.
> 
> ```
> The 28 channels limit applies to the sum of channels across all [Audio Element ]s in a [Mix Presentation ])before mixing.
> One example is a mix with 3rd-order Ambisonics (16 channels) + 7.1.4ch (12 channels).
> ```
> 
> if 9.1.6 layout is added as input layout , the max channels in a mix should be 32 which 16 channel is for 3rd-order Ambisonics and 16 channels are for 9.1.6

You are right. But as our latest consensus limits the sum of channels up to 28, it does not allow the mix of TOA and 9.1.6ch.

File: index.bs [hunk 6]
```diff
@@ -2438,6 +2463,9 @@ This section defines the renderer to use, given a channel-based [=Audio Element=
 ##### Rendering Without Demixing Info ##### {#processing-mixpresentation-rendering-m2l-withoutdemixinfo}
 - If the playback layout is neither 3.1.2ch nor 7.1.2ch,
     - If the playback layout complies with the loudspeaker layouts supported by [[!ITU-2051-3]], the EAR Direct Speakers renderer ([[ITU-2127-0]]) can be used, for example.
+    - Else if the playback layout is 9.1.6ch, 
+    	- If the input layout is [=22.2ch=], the down-mix matrix specified in [[#processing-downmixmatrix-static]] can be used, for example.
+    	- Else, the EAR Direct Speakers renderer ([[ITU-2127-0]]) can be used, for example, to first render the input audio to [=22.2ch=], followed by down-mixing from [=22.2ch=] to [=9.1.6ch=] by using the down-mix matrix specified in [[#processing-downmixmatrix-static]].
```
Comment (sunghee-hwang) [inline index.bs:2468]:
```suggestion
    	- Else, the EAR Direct Speakers renderer ([[ITU-2127-0]]) can be used, for example, to first render the input audio to [=22.2ch=], followed by copying LFE1 to LFE2 and followed by down-mixing from [=22.2ch=] to [=9.1.6ch=] by using the down-mix matrix specified in [[#processing-downmixmatrix-static]].
```

**Review by felicialim — APPROVED** [review]

**Review by felicialim — APPROVED** [review]

**Review by tdaede — APPROVED** [review]

**Pull request merged**